versiondb/memiavl/persisted_node.go

```diff
@@ -0,0 +1,139 @@
+package memiavl
+
+import (
+	"encoding/binary"
+	"os"
+
+	"github.com/ledgerwatch/erigon-lib/mmap"
+)
+
+const (
+	OffsetHeight  = 0
+	OffsetVersion = OffsetHeight + 4
+	OffsetSize    = OffsetVersion + 4
+	OffsetKey     = OffsetSize + 8
+	OffsetRight   = OffsetKey + 8
+	OffsetLeft    = OffsetRight + 4
+	OffsetValue   = OffsetKey + 8
+	OffsetHash    = OffsetValue + 8
+
+	SizeHash = 32
+	SizeNode = OffsetHash + SizeHash
+)
+
+type PersistedBlobs struct {
+	nodesFile  *os.File
+	keysFile   *os.File
+	valuesFile *os.File
+
+	nodes  []byte
+	keys   []byte
+	values []byte
+
+	// mmap handle for windows (this is used to close mmap)
+	nodesHandle  *[mmap.MaxMapSize]byte
+	keysHandle   *[mmap.MaxMapSize]byte
+	valuesHandle *[mmap.MaxMapSize]byte
+}
+
+func (blobs *PersistedBlobs) Node(index uint64) PersistedNode {
+	return PersistedNode{
+		blobs:  blobs,
+		offset: index * SizeNode,
+	}
+}
+
+func (blobs *PersistedBlobs) Close() error {
+	_ = mmap.Munmap(blobs.nodes, blobs.nodesHandle)
+	_ = mmap.Munmap(blobs.keys, blobs.keysHandle)
+	_ = mmap.Munmap(blobs.values, blobs.valuesHandle)
+
+	err1 := blobs.nodesFile.Close()
+	err2 := blobs.keysFile.Close()
+	err3 := blobs.valuesFile.Close()
+	*blobs = PersistedBlobs{}
+	if err1 != nil {
+		return err1
+	}
+	if err2 != nil {
+		return err2
+	}
+	return err3
```

Comment:
trying to close 3 files but here might return err1/err2 but other errors happen too.

Reply:
done, `hashicorp/go-multierror` is good at this